internal/wire/ietf_header.go

```diff
@@ -0,0 +1,144 @@
+package wire
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/internal/utils"
+)
+
+// The Header is the header of a QUIC Packet.
+// TODO: add support for the key phase
+type Header struct {
+	Type             uint8
```

Comment:
Should this be an enum?

Reply:
Yes. It will become a `protocol.PacketType` in #902.